src/v/kafka/requests/consumer_records.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/requests/consumer_records.h"
+
+#include "kafka/requests/kafka_batch_adapter.h"
+#include "model/record.h"
+
+namespace kafka {
+
+namespace {
+
+/// \brief A subset of the model::record_batch_header
+///
+/// Just enough to delineate record_batch within consumer_records.
+struct record_batch_info {
+    model::offset base_offset;
+    int32_t record_count;
+    int32_t size_bytes;
+    int32_t record_bytes() {
+        return size_bytes - model::packed_record_batch_header_size;
+    }
+    model::offset last_offset() {
+        return base_offset + model::offset{record_count - 1};
+    }
+};
+
+record_batch_info read_record_batch_info(iobuf_const_parser& in) {
+    const size_t initial_bytes_consumed = in.bytes_consumed();
+    const auto base_offset = model::offset(in.consume_be_type<int64_t>());
+    const auto batch_length = in.consume_be_type<int32_t>();
+    constexpr size_t skip_len_1 = sizeof(int32_t) + // partition_leader_epoch
+                                  sizeof(int8_t) +  // magic
+                                  sizeof(int32_t) + // crc
+                                  sizeof(int16_t);  // attrs
+
+    in.skip(skip_len_1);
+    const auto last_offset_delta = in.consume_be_type<int32_t>();
+    constexpr size_t skip_len_2 = sizeof(int64_t) + // first_timestamp
```

Comment:
should we give these meaningful names and push at the top of the impl file?

Reply:
I struggled to find meaningful names, and I like the comments naming the fields in order.

I think it might be best to have a constexpr type that describes the field type and offset, use it to define the fields that are required, and then little reader util that automatically skips from its present position to the required one,(I think it's ok to assume the fields are read in increasing order)

But then I thought I might as well wait for the new serialization design.